modules/PanelApp.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+"""Evidence parser for the Genomics England PanelApp data."""
+
+import argparse
+import logging
+import os
+import re
+import tempfile
+
+import requests
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import (
+    array, array_distinct, col, collect_set, concat, explode, lit, regexp_extract, regexp_replace, split, trim, when
+)
+
+
+class PanelAppEvidenceGenerator:
+
+    # Fixes which are applied to original PanelApp phenotype strings *before* splitting them by semicolon.
+    PHENOTYPE_BEFORE_SPLIT_RE = {
+        # Fixes for specific records.
+        r'\(HP:0006574;\);': r'(HP:0006574);',
+        r'Deafness, autosomal recessive; 12': r'Deafness, autosomal recessive, 12',
+        r'Waardenburg syndrome, type; 3': r'Waardenburg syndrome, type 3',
+        r'Ectrodactyly, ectodermal dysplasia, and cleft lip/palate syndrome; 3':
+            r'Ectrodactyly, ectodermal dysplasia, and cleft lip/palate syndrome, 3',
+
+        # Remove curly braces. They are *sometimes* (not consistently) used to separate disease name and OMIM code, for
+        # example: "{Bladder cancer, somatic}, 109800", and interfere with regular expressions for extraction.
+        r'[{}]': r'',
+
+        # Fix cases like "Aarskog-Scott syndrome, 305400Mental retardation, X-linked syndromic 16, 305400", where
+        # several phenotypes are glued to each other due to formatting errors.
+        r'(\d{6})([A-Za-z])': r'$1;$2',
+
+        # Replace all tab/space sequences with a single space.
+        r'[\t ]+': r' ',
+    }
+
+    # Cleanup regular expressions which are applied to the phenotypes *after* splitting.
+    PHENOTYPE_AFTER_SPLIT_RE = (
+        r' \(no OMIM number\)',
+        r' \(NO phenotype number in OMIM\)',
+        r'(No|no) OMIM (phenotype|number|entry)',
+        r'[( ]*(from )?PMID:? *\d+[ ).]*'
+    )
+
+    # Regular expressions for extracting ontology information.
+    LEADING = r'[ ,-]*'
+    SEPARATOR = r'[:_ #]*'
+    TRAILING = r'[:.]*'
+    OMIM_RE = LEADING + r'(OMIM|MIM)?' + SEPARATOR + r'(\d{6})' + TRAILING
+    OTHER_RE = LEADING + r'(OrphaNet: ORPHA|Orphanet|ORPHA|HP|MONDO)' + SEPARATOR + r'(\d+)' + TRAILING
+
+    # Regular expressions for extracting publication information from the API raw strings.
+    PMID_RE = [
+        (
+            r'^'                # Start of the string.
+            r'[\d, ]+'          # A sequence of digits, commas and spaces.
+            r'(?: |$)'          # Ending either with a space, or with the end of the string.
+        ),
+        (
+            r'(?:PubMed|PMID)'  # PubMed or a PMID prefix.
+            r'[: ]*'            # An optional separator (spaces/colons).
+            r'[\d, ]'           # A sequence of digits, commas and spaces.
```

Comment:
```suggestion
            r'[\d, ]+'           # A sequence of digits, commas and spaces.
```
I think the number of digits should be more than one, as it causes side effects. See my other comment on row 190. (I think it would be a good idea to show one-one typical examples the patterns are matching)

Reply:
Yes, this was exactly the problem. Fix committed in https://github.com/opentargets/evidence_datasource_parsers/commit/ac83edb4f832f10552faaff10e96eb036e53f1a0, and examples are added in https://github.com/opentargets/evidence_datasource_parsers/pull/88/commits/d49cc8440396258d271e63ca04d71f8bf211a661.

Also, all original + parsed PMIDs are available here: https://docs.google.com/spreadsheets/d/1otdZXXEmyNaLT7j67DwgHUM-pHPJaYc0SQnyqb1bUtA/edit?usp=sharing.